icu4c/source/common/unicode/umachine.h

```diff
@@ -52,6 +52,16 @@
 #include <stdbool.h>
 #include <stddef.h>
 
+#if U_CPLUSPLUS_VERSION == 0
+#if !U_PLATFORM_IS_DARWIN_BASED
+#include <uchar.h>
+#else
+// The <uchar.h> file does not exist in Apple SDKs (as of 3/15/24),
+// so we have to define char16_t ourselves
+typedef uint16_t char16_t;
+#endif // !U_PLATFORM_IS_DARWIN_BASED
```

Comment:
praise: I like this solution to Apple compilers having this gap in C11 support.

Reply:
Thank you!